sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -561,6 +561,128 @@ def fetch_metric(self, rows, columns):
+        FROM sys.dm_io_virtual_file_stats(DB_ID(), NULL) fs
+            LEFT JOIN sys.database_files df
+                ON df.file_id = fs.file_id;
+    """
+
+    def __init__(self, cfg_instance, base_name, report_function, column, logger):
+        super(SqlVirtualFileIOStats, self).__init__(cfg_instance, base_name, report_function, column, logger)
+
+    @classmethod
+    def fetch_all_values(cls, cursor, counters_list, logger, databases=None):
+        rows = []
+        columns = []
+
+        # Exclude the master database because sys.database_files is not available there
+        databases = [d for d in databases if d != 'master']
+
+        if not databases:
+            # No database list; use the server-wide master files DMV
+            query = cls.QUERY_MASTER_FILES
+
+            cursor.execute(query)
+            rows = cursor.fetchall()
+            columns = [i[0] for i in cursor.description]
+            logger.debug("%s: received %d rows and %d columns", cls.__name__, len(rows), len(columns))
+
+        else:
+            # Query each database in the list for the files in the current database
+            query = cls.QUERY_DATABASE_FILES
+
+            cursor.execute('select DB_NAME()')  # This can return None in some implementations so it cannot be chained
+            data = cursor.fetchall()
+            current_db = data[0][0]
+            logger.debug("%s: current db is %s", cls.__name__, current_db)
+
+            for db in databases:
+                # use statements need to be executed separate from select queries
+                ctx = construct_use_statement(db)
+                logger.debug("%s: changing cursor context via use statement: %s", cls.__name__, ctx)
+                cursor.execute(ctx)
+                logger.debug("%s: fetch_all executing query: %s", cls.__name__, query)
+                cursor.execute(query)
+                data = cursor.fetchall()
+
+                columns = [i[0] for i in cursor.description]
+                for r in data:
+                    rows.append(r)
+
+                logger.debug("%s: received %d rows and %d columns for db %s", cls.__name__, len(data), len(columns), db)
+
+            # reset back to previous db
+            logger.debug("%s: reverting cursor context via use statement to %s", cls.__name__, current_db)
+            cursor.execute(construct_use_statement(current_db))
+
+        return rows, columns
+
+    def fetch_metric(self, rows, columns):
+        db_name = columns.index('database_name')
+        state_desc = columns.index('state_desc')
+        logical_name = columns.index('logical_name')
+        physical_name = columns.index('physical_name')
+        column_val = columns.index(self.column)
+
+        for row in rows:
+            val = row[column_val]
+            if column_val is None:
+                continue
+
+            metric_tags = [
+                'database:{}'.format(row[db_name]),
```

Comment:
is `database` the tag that we use everywhere else for sqlserver metrics? why do we use `database` here, but `db` for our other DBM stuff?

Reply:
Good question. The standards appear to be scattered 🙃 

Do we align with DBM or other metrics?

Reply:
lol yeah, ugh. I think we could make everything align with DBM (as a follow up)? 

Not sure how wide-spread the damage would be for other non-DBM customers, but I think we should fix this since its confusing

Reply:
I updated the tag to be `db`